Lib/ListSelector/ListSelector.xaml.cs

```diff
@@ -177,18 +217,18 @@ private void SelectedListBox_OnMouseDoubleClick( object sender, MouseButtonEvent
 
         private static void OnAvailableItemsSourcePropertyChangedEvent( DependencyObject d, DependencyPropertyChangedEventArgs e )
         {
-            ( (ListSelector) d ).OnAvailableItemsSourcePropertyChangedEvent( (IEnumerable) e.OldValue, (IEnumerable) e.NewValue );
+            ( (ListSelector) d ).OnAvailableItemsSourcePropertyChangedEvent( (IEnumerable) e.OldValue, (IEnumerable?) e.NewValue );
```

Comment:
Good point in taking in account that e.NewValue may be null, but this should be also be considered for e.OldValue.

In fact, just handling nulls in the called function should be enough, maybe the (IEnumerable?) cast here is unnecessary.